--- lp_solve_4.0/Makefile.orig	Sun Nov 17 23:37:03 2002
+++ lp_solve_4.0/Makefile	Sun Dec  8 14:26:57 2002
@@ -9,7 +9,7 @@
 #CFLAGS= -g -Aa -D_POSIX_SOURCE -DCHECK +FP VZOUiD
 
 # nice for gcc
-CFLAGS= -O3 -Wall -pedantic -ansi
+CFLAGS= $(OPT) -Wall -pedantic -ansi
 #CFLAGS= -g -Wall -pedantic -ansi
 
 # Option -DCHECK checks for numerical problems during rounding of numbers.
@@ -41,6 +41,12 @@
 #non-ANSI math lib, should also work
 MATHLIB= -lm
 
+prefix = /usr
+bindir = $(prefix)/bin
+libdir = $(prefix)/lib
+includedir = $(prefix)/include/lp_solve
+mandir = $(prefix)/share/man
+
 LPKSRC.c= lpkit.c solve.c debug.c read.c readmps.c hash.c presolve.c
 LEXFILE.l= lex.l
 YACCFILE.y= lp.y
@@ -48,42 +54,44 @@
 TESTFILES2= lp_examples/*.mps
 
 TARGET=lp_solve
-LPKLIB=liblpk.a
+LPKLIB=liblpk.la
 
 LEXFILE.c= $(LEXFILE.l:.l=.c)
 YACCFILE.c= $(YACCFILE.y:.y=.c)
-YACCFILE.o= $(YACCFILE.y:.y=.o)
+YACCFILE.o= $(YACCFILE.y:.y=.lo)
 CSOURCES=lpkit.c solve.c debug.c read.c readmps.c lp_solve.c demo.c hash.c presolve.c $(LEXFILE.c) $(YACCFILE.c)
-COBJ=$(CSOURCES:.c=.o)
+COBJ=$(CSOURCES:.c=.lo)
 LPKSRC= $(LPKSRC.c) $(YACCFILE.c)
-LPKOBJ= $(LPKSRC:.c=.o)
+LPKOBJ= $(LPKSRC:.c=.lo)
 HEADERS=lpkit.h lpglob.h patchlevel.h debug.h read.h hash.h declare.h ufortify.h fortify.h
 
 all:	demo $(TARGET) lp2mps mps2lp
 
 $(COBJ): $(HEADERS)
 
+%.lo: %.c
+	libtool $(CC) $(CFLAGS) -c $< -o $@
+
 demo:	demo.o $(LPKLIB)
-	$(CC) -o demo $(CFLAGS) demo.o $(LPKLIB) $(LEXLIB) $(MATHLIB)
+	libtool $(CC) -o demo $(CFLAGS) demo.o $(LPKLIB) $(LEXLIB) $(MATHLIB)
 
 lp2mps:	lp2mps.o $(LPKLIB)
-	$(CC) -o lp2mps $(CFLAGS) lp2mps.o $(LPKLIB) $(LEXLIB) $(MATHLIB)
+	libtool $(CC) -o lp2mps $(CFLAGS) lp2mps.o $(LPKLIB) $(LEXLIB) $(MATHLIB)
 
 mps2lp: mps2lp.o $(LPKLIB)
-	$(CC) -o mps2lp $(CFLAGS) mps2lp.o $(LPKLIB) $(LEXLIB) $(MATHLIB)
+	libtool $(CC) -o mps2lp $(CFLAGS) mps2lp.o $(LPKLIB) $(LEXLIB) $(MATHLIB)
 
 $(TARGET): lp_solve.o $(LPKLIB)
-	$(CC) -o $(TARGET) $(CFLAGS) lp_solve.o $(LPKLIB) $(LEXLIB) $(MATHLIB)
+	libtool $(CC) -o $(TARGET) $(CFLAGS) lp_solve.o $(LPKLIB) $(LEXLIB) $(MATHLIB)
 
 purify: lp_solve.o $(LPKLIB)
-	purify $(CC) -o $(TARGET) $(CFLAGS) lp_solve.o $(LPKLIB) $(LEXLIB) $(MATHLIB)
+	purify libtool $(CC) -o $(TARGET) $(CFLAGS) lp_solve.o $(LPKLIB) $(LEXLIB) $(MATHLIB)
 
 quantify: lp_solve.o $(LPKLIB)
-	quantify $(CC) -o $(TARGET) $(CFLAGS) lp_solve.o $(LPKLIB) $(LEXLIB) $(MATHLIB)
+	quantify libtool $(CC) -o $(TARGET) $(CFLAGS) lp_solve.o $(LPKLIB) $(LEXLIB) $(MATHLIB)
 
 $(LPKLIB): $(LPKOBJ)
-	ar rv $@ $(LPKOBJ)
-	ranlib $(LPKLIB)
+	libtool $(CC) -o $(LPKLIB) $(LPKOBJ) -rpath $(libdir)
 
 $(YACCFILE.o): $(LEXFILE.c)
 
@@ -136,3 +144,12 @@
 
 TAGS:
 	etags *.[chyl]
+
+install: all
+	install -d $(DESTDIR)$(libdir) $(DESTDIR)$(includedir) $(DESTDIR)$(mandir)/man1 $(DESTDIR)$(bindir)
+	libtool install $(LPKLIB) $(DESTDIR)$(libdir)
+	libtool install $(TARGET) $(DESTDIR)$(bindir)
+	libtool install lp2mps $(DESTDIR)$(bindir)
+	libtool install mps2lp $(DESTDIR)$(bindir)
+	install $(HEADERS) $(DESTDIR)$(includedir)
+	install $(TARGET).1 $(DESTDIR)$(mandir)/man1
--- lp_solve_4.0/lpkit.h.orig	Sat Nov 16 21:25:50 2002
+++ lp_solve_4.0/lpkit.h	Sun Dec  8 16:50:22 2002
@@ -840,12 +840,7 @@
 short SOS_is_satisfied(lprec *lp, int sosindex, REAL *solution);
 MYBOOL SOS_can_mark(lprec *lp, int sosindex, int column);
 
-/* define yyparse() to make compilers happy. There should be some system
-   include file for this */
-int yyparse(void);
-void yyerror(char *);
 void check_decl(int);
-int yywrap();
 
 #ifdef __cplusplus
 };
--- lp_solve_4.0/lp.y.orig	Sun Sep 29 23:21:28 2002
+++ lp_solve_4.0/lp.y	Sun Dec  8 16:53:00 2002
@@ -13,6 +13,9 @@
 #include "lpglob.h"
 #include "read.h"
 
+void yyerror(char *);
+int yywrap();
+
 /* globals */
 char Last_var[NAMELEN];
 int Lin_term_count;
--- lp_solve_4.0/read.c.orig	Sun Nov 17 23:36:04 2002
+++ lp_solve_4.0/read.c	Sun Dec  8 16:53:53 2002
@@ -35,6 +35,8 @@
 #include "lpkit.h"
 #include "lpglob.h"
 
+int yyparse(void);
+
 short            *relat;
 int              Verbose;
 constraint_name  *First_constraint_name;
